### PR TITLE
Add tile logic UWP to allow custom tiles

### PR DESCRIPTION
**Pull request by JKennedy24**
Added Tile Logic to UWP to allow custom tile sources

### CODE REVIEW

**Comment by JKennedy24** [comment]
I think this one may have gone wrong, I needed the code from the Pin Logic refactor. But I think I might have branched from Feature_branch. Should I have branched from the Pin_Logic branch and added the MapTileLogic code to that?

File: Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.UWP/Logics/PinLogic.cs [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+
+
+        internal override void Unregister(MapControl nativeMap, Map map)
+        {
+            if (nativeMap != null)
+            {
+                nativeMap.MapTapped -= NewNativeMap_MapTapped;
+                nativeMap.MapHolding -= NewNativeMap_MapHolding;
+                //newNativeMap.MarkerDragStart += OnMarkerDragStart;
```
Comment (deextercz) [inline Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.UWP/Logics/PinLogic.cs]: Please remove commented lines or create event even if it is not handled

File: Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.UWP/Logics/PinLogic.cs [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    internal class PinLogic : DefaultPinLogic<PushPin, MapControl>
+    {
+        internal override void Register(MapControl oldNativeMap, Map oldMap, MapControl newNativeMap, Map newMap)
+        {
+            base.Register(oldNativeMap, oldMap, newNativeMap, newMap);
+            if (newNativeMap != null)
+            {
+                newNativeMap.MapTapped += NewNativeMap_MapTapped;
+                newNativeMap.MapHolding += NewNativeMap_MapHolding;
+                //newNativeMap.MarkerDragStart += OnMarkerDragStart;
```
Comment (deextercz) [inline Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.UWP/Logics/PinLogic.cs]: Please remove commented lines or create event even if it is not handled

File: Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.UWP/MapRenderer.cs [hunk 3]
```diff
@@ -58,12 +86,19 @@ public class MapRenderer : ViewRenderer<Map, MapControl>
                 UpdateHasScrollEnabled();
                 UpdateHasZoomEnabled();
 
-                ((ObservableCollection<Pin>)mapModel.Pins).CollectionChanged += OnCollectionChanged;
+                //((ObservableCollection<Pin>)mapModel.Pins).CollectionChanged += OnCollectionChanged;
```
Comment (deextercz) [inline Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.UWP/MapRenderer.cs]: Remove commented code

File: Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.UWP/MapRenderer.cs [hunk 4]
```diff
@@ -28,6 +30,31 @@ namespace Xamarin.Forms.Maps.WinRT
...
+            get { return Control as MapControl; }
+        }
+
+        readonly BaseLogic<MapControl>[] _logics;
+
+        public MapRenderer() : base()
+        {
+            _logics = new BaseLogic<MapControl>[]
+            {
+                //new PolylineLogic(),
```
Comment (deextercz) [inline Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.UWP/MapRenderer.cs]: Remove commented code

File: Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.UWP/MapRenderer.cs [hunk 5]
```diff
@@ -99,63 +134,63 @@ protected override void Dispose(bool disposing)
...
-
-            if (pushPin != null)
-                Control.Children.Remove(pushPin);
-        }
-
-        void LoadPin(Pin pin)
-        {
-            Control.Children.Add(new PushPin(pin));
-        }
+//        void OnCollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
```
Comment (deextercz) [inline Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps.UWP/MapRenderer.cs]: Remove commented code

**Comment by amay077** [comment]
Hi @deextercz ,
Yes I think so.

**Comment by JKennedy24** [comment]
I have removed all the commented out code as @deextercz suggested. If someone could exclude the files I have accidentally included here such as project files (apart from GoogleMaps.UWP) then that would be great

**Comment by deextercz** [comment]
Please solve conflicts first, I created pull request to my repository https://github.com/JKennedy24/Xamarin.Forms.GoogleMaps/pull/1 but it is not possible until merge conflicts will be solved.

**Comment by deextercz** [comment]
I fixed your pullrequest and create new one.
https://github.com/amay077/Xamarin.Forms.GoogleMaps/pull/136/files
Please review it and test it.

**Comment by JKennedy24** [comment]
@deextercz Thanks for your help, That looks perfect! I will now close this pull request off

**Pull request closed without merging**